### PR TITLE
Abstraction! Liberation!

### PR DESCRIPTION
**Pull request by Draycia**
Prepares the plugin to be multi platform.

- [x] Move API related stuff to CarbonChat-API
- [x] Move common / core stuff to CarbonChat-Common

Platforms:
- [x] CarbonChat-Bukkit

Other stuff:
- [x] Switch to Cloud for commands.
- [x] Switch to configurate for configs. 
- [x] Command settings loading
- [x] Channel loading
- [ ] Platform specific configs / options
- [x] Ensure all options are present and functional
- [x] Whisper overhaul

### CODE REVIEW

File: CarbonChat-API/src/main/java/net/draycia/carbon/api/channels/ChatChannel.java [hunk 1]
```diff
@@ -15,46 +14,49 @@
 
 public abstract class ChatChannel implements ForwardingAudience {
```
Comment (Sauilitired) [inline CarbonChat-API/src/main/java/net/draycia/carbon/api/channels/ChatChannel.java:15]: Why isn't ChatChannel an interface? It contains no state, and the one implemented method could just as well be a default interface method

Comment (Draycia) [reply CarbonChat-API/src/main/java/net/draycia/carbon/api/channels/ChatChannel.java]: 168233d

**Comment by Laarryy** [comment]
This is a silly pull request title. and on your own project. tsk tsk tsk

File: CarbonChat-API/src/main/java/net/draycia/carbon/api/channels/WhisperChannel.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  /**
+   * The recipient for this channel, where messages are sent to.
+   * @return The channel recipient.
+   */
+  @Override
+  @NonNull ChatUser audience();
+
+  @Override
+  default @NonNull String name() {
+    return "Party";
```
Comment (broccolai) [inline CarbonChat-API/src/main/java/net/draycia/carbon/api/channels/WhisperChannel.java:20]: Why is the default for this party, if it's not a comment on the function explaining why it is would be nice

File: CarbonChat-API/src/main/java/net/draycia/carbon/api/events/UserEvent.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+public class UserEvent implements CarbonEvent, Cancellable {
+
+  private final @NonNull ChatUser user;
+  private boolean cancelled = false;
+
+  public UserEvent(final @NonNull ChatUser user) {
+    this.user = user;
+  }
+
```
Comment (broccolai) [inline CarbonChat-API/src/main/java/net/draycia/carbon/api/events/UserEvent.java:16]:
```suggestion
@Nonnull
```
Missing nullability

File: CarbonChat-Bukkit/build.gradle [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  compile 'net.kyori:adventure-platform-bukkit:4.0.0-SNAPSHOT'
+
+  compile project(":CarbonChat-Common")
+
+  // Messaging
+  compile 'io.github.leonardosnt:bungeechannelapi:1.0.0-SNAPSHOT'
+
+  // Commands
+  compile "com.intellectualsites:cloud-paper:1.0-SNAPSHOT"
+
```
Comment (broccolai) [inline CarbonChat-Bukkit/build.gradle:20]: implementation / api instead of compile

Comment (zml2008) [reply CarbonChat-Bukkit/build.gradle:20]: yeah, `compile` is gonna go away in gradle 7

File: CarbonChat-Common/src/main/java/net/draycia/carbon/common/adventure/AdventureManager.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  @NonNull
+  private final AudienceProvider provider;
+
+  @NonNull
+  private final CarbonChat carbonChat;
+
+  @NonNull
+  private final FormatType formatType;
+
+  public AdventureManager(final @NonNull AudienceProvider provider, final @NonNull FormatType formatType) {
```
Comment (broccolai) [inline CarbonChat-Common/src/main/java/net/draycia/carbon/common/adventure/AdventureManager.java:24]: Different ordering of nullability and final(ity?)

**Review by broccolai — COMMENTED** [review]
Only a few minor things with mainly formatting, implementations look good though.
I've left out parts that had comments already or a todo, also I think TextChannel is a fine name for what it represents.

**Pull request merged**